src/submission/NodeSubmissionClient.ts

```diff
@@ -16,44 +16,35 @@ export class NodeSubmissionClient extends DefaultSubmissionClient {
     this.configurationVersionHeader = this.configurationVersionHeader.toLowerCase();
   }
 
-  public sendRequest(config:Configuration, method:string, path:string, data:string, callback: (status:number, message:string, data?:string, headers?:Object) => void): void {
-    function complete(response:http.IncomingMessage, responseBody:string, responseHeaders:Object) {
-      var message:string;
-      if (response.statusCode === 0) {
-        message = 'Unable to connect to server.';
-      } else if (response.statusCode < 200 || response.statusCode > 299) {
-        message = response.statusMessage || (<any>response).message;
-      }
+  public sendRequest(config: Configuration, method: string, path: string, data: string, callback: NodeSubmissionCallback): void {
```

Comment:
This overrides the base method if I'm following correctly, I'm not sure how this compiles as we are changing the callback signature from everything that's expecting it. Ahh, I see you made this much nicer and into an interface. Do you think perhaps we should make this non node specific and update the default submission client too? I like what you did here :+1: